flumine/markets/middleware.py

```diff
@@ -156,30 +159,93 @@ def _process_runner_removal(
                                 ),
                                 extra=order.info,
                             )
-                    elif (
-                        removal_adjustment_factor
-                        and removal_adjustment_factor >= WIN_MINIMUM_ADJUSTMENT_FACTOR
-                    ):
-                        # todo place market
-                        for match in order.simulated.matched:
-                            match[1] = self._calculate_reduction_factor(
-                                match[1], removal_adjustment_factor
+                    elif removal_adjustment_factor:
+                        if (market.market_type == "WIN") and (
+                            removal_adjustment_factor >= WIN_MINIMUM_ADJUSTMENT_FACTOR
+                        ):
+                            for match in order.simulated.matched:
+                                match[1] = self._calculate_reduction_factor(
+                                    match[1], removal_adjustment_factor
+                                )
+                            _, order.simulated.average_price_matched = wap(
+                                order.simulated.matched
+                            )
+
+                            if order.side == "LAY":
+                                if (
```

Comment:
Don't we already handle this on line 117?

Reply:
I don't think so.

On line 174, the Order Type will be LIMIT_ON_CLOSE or LIMIT if the order's a LAY.

On line 117, the Order Type is MARKET_ON_CLOSE for LAY bets.